hazelcast/src/test/java/com/hazelcast/jet/pipeline/SinksTest.java

```diff
@@ -276,6 +283,153 @@ public void remoteMap() {
+    }
+
+    @Test
+    public void remoteReplicatedMap_customBatchSize() {
+        remoteReplicatedMap(() -> Sinks.remoteReplicatedMap(sinkName, clientConfig, 100));
+    }
+
+    private void remoteReplicatedMap(Supplier<Sink<Entry<String, Integer>>> sinkSupplier) {
+        // Given
+        List<Integer> input = sequence(itemCount);
+        BatchSource<Integer> source1 = TestSources.items(input);
+        // When
+        Sink<Entry<String, Integer>> sink = sinkSupplier.get();
+        p.readFrom(source1)
+                .map(i -> entry(String.valueOf(i), i))
+                .writeTo(sink);
+        execute();
+        // Then
+        Map<String, Integer> expected = sequence(itemCount).stream()
+                .collect(toMap(String::valueOf, i -> i));
+
+        Map<String, Integer> actual = remoteHz.getReplicatedMap(sinkName);
+        assertThat(actual).containsExactlyInAnyOrderEntriesOf(expected);
+    }
+
+    @Test
+    public void remoteReplicatedMap_dataConnectionName() {
+        remoteReplicatedMap_dataConnectionName((dataConnectionName) -> Sinks.remoteReplicatedMap(sinkName, dataConnectionName));
+    }
+
+    @Test
+    public void remoteReplicatedMap_dataConnectionName_customBatchSize() {
+        remoteReplicatedMap_dataConnectionName((dataConnectionName) -> Sinks.remoteReplicatedMap(sinkName, dataConnectionName, 100));
+    }
+
+    private void remoteReplicatedMap_dataConnectionName(Function<String, Sink<Entry<String, Integer>>> sinkFn) {
+        // Given
+        List<Integer> input = sequence(itemCount);
+        BatchSource<Integer> source1 = TestSources.items(input);
+        // When
+        String dataConnectionName = randomString();
+        hz().getConfig().addDataConnectionConfig(new DataConnectionConfig(dataConnectionName)
+                .setType("Hz")
+                .setShared(false)
+                .setProperty(HazelcastDataConnection.CLIENT_XML, ImdgUtil.asXmlString(clientConfig)));
+        Sink<Entry<String, Integer>> sink = sinkFn.apply(dataConnectionName);
+        p.readFrom(source1)
+                .map(i -> entry(String.valueOf(i), i))
+                .writeTo(sink);
+        execute();
+        // Then
+        Map<String, Integer> expected = sequence(itemCount).stream()
+                                                           .collect(toMap(String::valueOf, i -> i));
+
+        Map<String, Integer> actual = remoteHz.getReplicatedMap(sinkName);
+        assertThat(actual).containsExactlyInAnyOrderEntriesOf(expected);
+    }
+
+    @Test
+    public void remoteReplicatedMap_nonExistentDataConnection() {
+        remoteReplicatedMap_nonExistentDataConnection((dataConnectionName) -> Sinks.remoteReplicatedMap(sinkName, dataConnectionName));
+    }
+
+    @Test
+    public void remoteReplicatedMap_nonExistentDataConnection_customBatchSize() {
+        remoteReplicatedMap_nonExistentDataConnection((dataConnectionName) -> Sinks.remoteReplicatedMap(sinkName, dataConnectionName, 100));
+    }
```

Comment:
This one actually seems redundant. Is there a code path this and only this test covers?